### PR TITLE
Feat: load the system version requirements from the addon's meta.yaml(#395)

### LINKED ISSUE #395: Feat: load the system version requirements from the addon's meta.yaml
A new feature of automatically finding available addon‘s version has been added to KubevVela's repository. 

> https://github.com/kubevela/kubevela/issues/4181

In order to adapt this feature, we need to add a `system` field to the `meta.yaml` file to record the system version requirements corresponding to a certain version of the addon. 

```yaml
name: example
version: 1.0.1
description: Extended workload to do continuous and progressive delivery
icon: https://raw.githubusercontent.com/fluxcd/flux/master/docs/_files/weave-flux.png
url: https://fluxcd.io
system: "vela>=1.4.0; kubernetes>=1.20.0"
```

This information will be loaded when synchronizing the addon package, and finally loaded to the `index.yaml` corresponding to the addon repository.

Can you provide better advice on where to load system version requirements? @wangyikewxgm  @wonderflow

### PR DESCRIPTION
**Pull request by Hanmengnan**
load the system version requirements from the addon's meta.yaml, and then write this value as a value to ChartVersions.Metadata.Annotations, the corresponding key is "system".

Signed-off-by: HanMengnan <1448189829@qq.com>

### Description of your changes

hack/addons/syn_addon_package.go

Fixes #395

A `System` field is added to the metadata structure in `hack/addons/syn_ addon_ package.go` to record the system version requirements.

To adapt this feature, we need to add a `system` field to addon's `meta.yaml` file.
```yaml
name: example
version: 1.0.1
description: Extended workload to do continuous and progressive delivery
icon: https://raw.githubusercontent.com/fluxcd/flux/master/docs/_files/weave-flux.png
url: https://fluxcd.io
system:
  vela: ">=v1.4.0-beta.2"
  kubernetes: ">=1.20.0"
```

### Checklist

I have:

- [ ] Title of the PR starts with type (e.g. `[Addon]` , `[example]` or `[Doc]`).
- [ ] Updated/Added any relevant [documentation](https://kubevela.io/docs/reference/addons/overview) and [examples](../examples).
- [ ] New addon should be put in [experimental](../experimental/addons).
- [ ] Update addon should modify the `version` in `metadata.yaml` to generate a new version.
- [x] Any changes about [verified](../addons) addons should be tested with CI [script](../test/e2e-test/hack/addon-vela-test.sh).

### CODE REVIEW

**Review by wangyikewxgm — CHANGES_REQUESTED** [review]
Please take a look this pr: https://github.com/kubevela/kubevela/pull/4284. You didn't considered one addon only have on type of system requirement before, I fixed it in it. Let's seperate to two annotation key-value.

**Comment by Hanmengnan** [comment]
> Please take a look this pr: [kubevela/kubevela#4284](https://github.com/kubevela/kubevela/pull/4284). You didn't consider one addon only have on type of system requirement before, I fix by it. Let's separate to two annotation key-value.

OK, i have separated to two annotation key-value.

**Review by wangyikewxgm — APPROVED** [review]
LGTM

File: hack/addons/syn_addon_package.go [hunk 1]
```diff
@@ -105,7 +111,7 @@ func main() {
 			entry := repo.ChartVersions{}
 			entry = append(entry, &repo.ChartVersion{Metadata: &chart.Metadata{Name: info.Name(),
 				Version: m.Version, Icon: m.Icon, Keywords: m.Tags, Description: m.Description,
-				Home: m.URL}, Created: time.Now(), URLs: []string{repoURL + "/" + info.Name() + "-" + m.Version + ".tgz"}})
+				Home: m.URL, Annotations: map[string]string{"system.vela": m.System.Vela, "system.kubernetes": m.System.Kubernetes}}, Created: time.Now(), URLs: []string{repoURL + "/" + info.Name() + "-" + m.Version + ".tgz"}})
```
Comment (wangyikewxgm) [inline hack/addons/syn_addon_package.go:114]: We should judge system filed if is empty.

**Review by wangyikewxgm — CHANGES_REQUESTED** [review]

File: hack/addons/syn_addon_package.go (hunk 1 above)
Comment (Hanmengnan) [reply hack/addons/syn_addon_package.go:114]: OK, I'll fix it here

**Review by wonderflow — APPROVED** [review]

**Pull request merged**